src/java/htsjdk/tribble/AbstractFeatureReader.java

```diff
@@ -122,35 +122,40 @@ protected AbstractFeatureReader(final String path, final FeatureCodec<T, SOURCE>
     /**
      * Whether the reader has an index or not
      * Default implementation returns false
+     *
      * @return
      */
-    public boolean hasIndex(){
+    public boolean hasIndex() {
         return false;
     }
 
-    public static void setComponentMethods(ComponentMethods methods){
+    public static void setComponentMethods(ComponentMethods methods) {
         AbstractFeatureReader.methods = methods;
     }
 
     /**
      * Whether a filename ends in one of the BLOCK_COMPRESSED_EXTENSIONS
-     * @param fileName
+     *
+     * @param path
      * @return
      */
-    public static boolean hasBlockCompressedExtension (final String fileName) {
+    public static boolean hasBlockCompressedExtension(final String path) {
```

Comment:
Add a unit test proving that the original issue with `hasBlockCompressedExtension()` is resolved.